src/kibana/components/vislib/lib/data.js

```diff
@@ -298,11 +307,20 @@ define(function (require) {
       var isOrdered = (this.data.ordered && this.data.ordered.date);
       var minDate = isOrdered ? this.data.ordered.min : undefined;
       var maxDate = isOrdered ? this.data.ordered.max : undefined;
+      var interval = isOrdered ? this.data.ordered.interval : undefined;
 
       return d3.max(series, function (data) {
         return d3.max(data, function (d) {
+
+          // Calculates the end time for each x value to compare against the
+          // minDate. If the start time of the bar is outside of the date range selected,
+          // but the end time is within the date range selected, the bar will still
+          // appear in the chart and thus its y value needs to be included in
+          // the y max value calculation so as not to cause bars going off the y scale.
+          var bucket = d.x + interval;
+
           if (isOrdered) {
```

Comment:
This is actually a good place to re-factor and reduce some redundant code.

```javascript
var calculateYMaxWithinDateRange = function (calculation) {
return function (series) {
      var isOrdered = (this.data.ordered && this.data.ordered.date);
      var minDate = isOrdered ? this.data.ordered.min : undefined;
      var interval = isOrdered ? this.data.ordered.interval : undefined;

      return d3.max(series, function (data) {
        return d3.max(data, function (d) {

          // Calculates the end time for each x value to compare against the
          // minDate. If the start time of the bar is outside of the date range selected,
          // but the end time is within the date range selected, the bar will still
          // appear in the chart and thus its y value needs to be included in
          // the y max value calculation so as not to cause bars going off the y scale.
          var bucket = d.x + interval;
          var outsideDateRange = bucket <= minDate;
          if (isOrdered && outsideDateRange) return;
          return calculation(d);
        });
      });
    };
}; 

Data.prototype.getYStackMax = calculateYMaxWithinDateRange(function (d) { return d.y0 + d.y });
Data.prototype.getYMax = calculateYMaxWithinDateRange(function (d) { return d.y });
```

After looking at this function is there ever a scenario where you would want to perform this calculation if the data is not ordered? The issue is that if the data is not ordered then interval is set to undefined. When you add that to an integer you will end up with NaN.

Soooo... I would write it this way:

```javascript
var calculateYMaxWithinDateRange = function (calculation) {
return function (series) {
      if (!this.data.ordered) return;
      var minDate = this.data.ordered.min;
      var interval = this.data.ordered.interval;

      return d3.max(series, function (data) {
        return d3.max(data, function (d) {

          // Calculates the end time for each x value to compare against the
          // minDate. If the start time of the bar is outside of the date range selected,
          // but the end time is within the date range selected, the bar will still
          // appear in the chart and thus its y value needs to be included in
          // the y max value calculation so as not to cause bars going off the y scale.
          if (d.x + interval <= minDate) return;
          return calculation(d);
        });
      });
    };
}; 

Data.prototype.getYStackMax = calculateYMaxWithinDateRange(function (d) { return d.y0 + d.y });
Data.prototype.getYMax = calculateYMaxWithinDateRange(function (d) { return d.y });
```